### PR TITLE
Use core M2

### PR DESCRIPTION
**Pull request by timyates**
Fix up removed and deprecated methods

### CODE REVIEW

File: servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java [hunk 1]
```diff
@@ -189,7 +189,7 @@ protected Object buildBody() {
     private Argument<?> resolveBodyType() {
         RouteMatch<?> route = this.getAttribute(HttpAttributes.ROUTE_MATCH, RouteMatch.class).orElse(null);
         if (route != null) {
-            Argument<?> bodyType = route.getBodyArgument()
+            Argument<?> bodyType = route.getRouteInfo().getFullBodyArgument()
```
Comment (timyates) [inline servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java:192]: @graemerocher Is this the correct replacement?

Comment (timyates) [reply servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java:192]: Seems not (or something else has changed) as now this is failing

https://github.com/micronaut-projects/micronaut-servlet/blob/f23fc5dd2b6cdd3ca25242c366176bf3e037d614/http-server-jetty/src/test/groovy/io/micronaut/servlet/jetty/JettyExceptionHandlerSpec.groovy#L44-L52

Comment (dstepanov) [reply servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java:192]: Looks correct

Comment (timyates) [reply servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java:192]: @dstepanov There's also `route.getRouteInfo().getBodyArgument()`... What's the difference?

**Comment by timyates** [comment]
So it seems the issue with the failing test:

https://github.com/micronaut-projects/micronaut-servlet/blob/f23fc5dd2b6cdd3ca25242c366176bf3e037d614/http-server-jetty/src/test/groovy/io/micronaut/servlet/jetty/JettyExceptionHandlerSpec.groovy#L44-L52

Is that when we execute normalFlow() for the request lifecycle here

https://github.com/micronaut-projects/micronaut-servlet/blob/a79b1c1c797c90bff6d0373bf55820a846576277/servlet-core/src/main/java/io/micronaut/servlet/http/ServletHttpHandler.java#L207-L211

We're getting a 200 response (with the expected body) wheras prior to core-M2, we got a 500

@dstepanov Can you spot what I've missed?  I'm assuming there's something new I need to set up in the RequestLifecycle?

**Comment by dstepanov** [comment]
@timyates To me, the test doesn't look correct. The exception is remapped so there shouldn't be any error. Maybe we can add that test to TCK?

**Pull request merged**